### PR TITLE
[Instrumentation.SqlClient] nullable enable

### PR DESCRIPTION
**Pull request by TimothyMothra**
Towards #894

## Changes
- remove `nullable disable` from SqlClient project and test project.

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs [hunk 1]
```diff
@@ -94,14 +94,20 @@ public override void OnEventWritten(string name, object payload)
...
+                        if (database != null)
+                        {
+                            activity.DisplayName = (string)database;
+                        }
 
                         _ = this.dataSourceFetcher.TryFetch(connection, out var dataSource);
                         _ = this.commandTextFetcher.TryFetch(command, out var commandText);
 
-                        activity.SetTag(SemanticConventions.AttributeDbName, (string)database);
+                        activity.SetTag(SemanticConventions.AttributeDbName, database);
```
Comment (Kielek) [inline src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs:105]: Not directly related to this PR.

Shouldn't it be in databased != statement? I think that we do not need empty/null attribute,

File: src/OpenTelemetry.Instrumentation.SqlClient/SqlClientTraceInstrumentationOptions.cs [hunk 2]
```diff
@@ -132,7 +132,7 @@ public class SqlClientTraceInstrumentationOptions
     /// cref="Activity"/>.</item>
     /// </list>
     /// </remarks>
-    public Action<Activity, string, object> Enrich { get; set; }
+    public Action<Activity, string, object>? Enrich { get; set; }
```
Comment (Kielek) [inline src/OpenTelemetry.Instrumentation.SqlClient/SqlClientTraceInstrumentationOptions.cs:135]: Just double checking if object (SqlCommand) is always non null here.

File: src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs (hunk 1 above)
Comment (TimothyMothra) [reply src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs:105]: good catch, I missed that.
I'm fine making this change in this PR, but let me know if this should be done in a follow up instead.

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs:105]: I would put it in scope of this pr

File: src/OpenTelemetry.Instrumentation.SqlClient/SqlClientTraceInstrumentationOptions.cs (hunk 2 above)
Comment (TimothyMothra) [reply src/OpenTelemetry.Instrumentation.SqlClient/SqlClientTraceInstrumentationOptions.cs:135]: I'm not sure on the requirements, but reviewing how Enrich is invoked:

`SqlClientDiagnosticListener` first checks if command is null before invoking.


https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/c34b5cd08ca9b81f04a85a0503984d7e479c4e67/src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs#L67-L72


https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/c34b5cd08ca9b81f04a85a0503984d7e479c4e67/src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs#L133

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.SqlClient/SqlClientTraceInstrumentationOptions.cs:135]: Thanks for confirmation

**Review by Kielek — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**